### ISSUE TITLE
Locale / Translation Error: The message file for category 'yii' does not exist

### ISSUE THREAD

**Issue by brucealdridge** [yiisoft/yii2#11671]
### What steps will reproduce the problem?
**Config:**
```
        'formatter' => [
            'locale' => 'en_AU',
        ]
```
**View:**
```
Yii::$app->formatter->asDate('');
```

### What is the expected result?
no errors

### What do you get instead?
error in logs
```
The message file for category 'yii' does not exist: /vagrant/www/vendor/yiisoft/yii2/messages/en_AU/yii.php Fallback file does not exist as well: /vagrant/www/vendor/yiisoft/yii2/messages/en/yii.php
/vagrant/www/views/default/index.php (8)
/vagrant/www/controllers/DefaultController.php (15)
```

### My Workaround
**Config:**
```
        'i18n' => [
            'translations' => [
                'yii' => [
                    'class' => 'yii\i18n\PhpMessageSource',
                    'basePath' => '@app/messages',
                ]
            ]
        ]
```
**@app/messages/en/yii.php**
```
   <?php return [];
```

Not sure if there is an error with my config/locale setup or if its a bug, docs aren't very clear other english locales. But my workaround seems like a hack.

**Comment by brucealdridge** [comment]
Easy fix is adding `<?php return [];`  to framework/messages/en/yii.php 

Is there a reason there isn't a `en/` folder? (Apart from being hacky)

**Comment by samdark** [comment]
Formatter locale should not affect this log message. Have you changed application `sourceLanguage`?

**Comment by brucealdridge** [comment]
Heres a barebones example.

https://github.com/brucealdridge/yii_11671

**Comment by samdark** [comment]
That's extremely weird. Thanks for complete test case.

**Comment by samdark** [comment]
Already fixed by #11429
